gui/storage/forms.py

```diff
@@ -1274,74 +1279,7 @@ class ZFSDataset(Form):
-        if self._create:
-            parentdata = _n.zfs_get_options(self._fs)
-        else:
-            parentdata = _n.zfs_get_options(self._fs.rsplit('/', 1)[0])
-
-        self.fields['dataset_atime'].choices = _inherit_choices(
-            choices.ZFS_AtimeChoices,
-            parentdata['atime'][0]
-        )
-        self.fields['dataset_compression'].choices = _inherit_choices(
-            choices.ZFS_CompressionChoices,
-            parentdata['compression'][0]
-        )
-        self.fields['dataset_dedup'].choices = _inherit_choices(
-            choices.ZFS_DEDUP_INHERIT,
-            parentdata['dedup'][0]
-        )
-
-        if self._create is False:
-            del self.fields['dataset_name']
-            del self.fields['dataset_recordsize']
-            del self.fields['dataset_case_sensitivity']
-            data = _n.zfs_get_options(self._fs)
-
-            if 'org.freenas:description' in data and data['org.freenas:description'][2] == 'local':
-                self.fields['dataset_comments'].initial = data['org.freenas:description'][0]
-
-            if data['compression'][2] == 'inherit':
-                self.fields['dataset_compression'].initial = 'inherit'
-            else:
-                self.fields['dataset_compression'].initial = data['compression'][0]
-            self.fields['dataset_share_type'].initial = _n.get_dataset_share_type(self._fs)
-
-            if data['atime'][2] == 'inherit':
-                self.fields['dataset_atime'].initial = 'inherit'
-            else:
-                self.fields['dataset_atime'].initial = data['atime'][0]
-
-            for attr in ('refquota', 'quota', 'reservation', 'refreservation'):
-                formfield = 'dataset_%s' % (attr)
-                if data[attr][0] == 'none':
-                    self.fields[formfield].initial = 0
-                else:
-                    self.fields[formfield].initial = data[attr][0]
-            if data['dedup'][2] == 'inherit':
-                self.fields['dataset_dedup'].initial = 'inherit'
-            elif data['dedup'][0] in ('on', 'off', 'verify'):
-                self.fields['dataset_dedup'].initial = data['dedup'][0]
-            elif data['dedup'][0] == 'sha256,verify':
-                self.fields['dataset_dedup'].initial = 'verify'
-            else:
-                self.fields['dataset_dedup'].initial = 'off'
-
-        if not dedup_enabled():
-            self.fields['dataset_dedup'].widget.attrs['readonly'] = True
-            self.fields['dataset_dedup'].widget.attrs['class'] = (
-                'dijitSelectDisabled dijitDisabled')
-            self.fields['dataset_dedup'].widget.text = mark_safe(
-                '<span style="color: red;">Dedup feature not activated. '
-                'Contact <a href="mailto:truenas-support@ixsystems.com?subject'
-                '=ZFS Deduplication Activation">TrueNAS Support</a> for '
-                'assistance.</span><br />'
-            )
```

Comment:
You have removed all these from the Create forms which seems wrong. It will no longer report the inherit value on create time nor show dedup warning.